velox/vector/tests/VectorPoolTest.cpp

```diff
@@ -151,4 +151,48 @@ TEST_F(VectorPoolTest, customTypes) {
   ASSERT_EQ(1'000, vector->size());
   ASSERT_TRUE(isJsonType(vector->type()));
 }
+
+TEST_F(VectorPoolTest, clear) {
+  const auto statsBefore = pool()->stats();
+
+  VectorPool vectorPool(pool());
+
+  std::vector<VectorPtr> vectors(10);
+  std::vector<BaseVector*> vectorPtrs(10);
```

Comment:
there is a very little but very real chance of the addresses coming out the same when we reallocate at L193 so there is a chance of occasional flakiness. Can you replace this with a vector of `std::weak_ptr` instead and then we can check whether they are no longer valid after calling `vectorPool.clear();`